src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+                    if (sunken){
+                        let shipLength:number = row-startRow;
+                        this.remainingShips = this.remove(shipLength);
+                        this.sink(startRow,column,row-1,column);
+                    }
+                }
+            }
+        }
+    }
+
+    public remove(shipLength:number):number[]{
+        let ans:number[] = [];
+        let done:boolean = false;
+        for (let i:number = 0; i<this.remainingShips.length; i++){
+            if (done){
+                ans.push(this.remainingShips[i]);
+            }
+            else {
+                if (this.remainingShips[i]!=shipLength){
+                    ans.push(this.remainingShips[i]);
+                }
+                else{
+                    done = true;
+                }
+            }
+        }
+        return ans;
+    }
+
+    public eliminateNeighboursOfSunken():void{
+        let x:number[] = [1,1,1,0,-1,-1,-1,0];
+        let y:number[] = [-1,0,1,1,1,0,-1,-1];
+        for (let row:number = 0; row < 10; row++){
+            for (let column:number = 0; column < 10; column ++){
+                if (this.board[row][column] == 2){
+                    for (let i:number = 0; i<8; i++){
+                        let dr:number = row + x[i];
+                        let dc:number = column + y[i];
+                        if (this.isValidTarget(new Position({"Row":GameState.numberToLetter[dr], "Column":dc+1}))){
+                            this.board[dr][dc] = -1;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public sink(startRow:number, startColumn:number, endRow:number, endColumn:number):void{
+        for (let row:number=startRow; row<=endRow; row++){
+            for (let column:number = startColumn; column <= endColumn; column++){
+                this.board[row][column] = 2;
+            }
+        }
+    }
+
+    public isSunken(startRow:number, startColumn:number, endRow:number, endColumn:number):boolean{
+        if ((startRow==endRow) && (startColumn == endColumn)){
+            return false;
+        }
+        if (startRow!=endRow){
+            if (Math.abs(endRow - startRow)+1 == this.remainingShips[this.remainingShips.length-1]){
+                return true;
+            }
+            if ((startRow == 0) || (this.board[startRow-1][startColumn]==-1)){
+                if ((endRow == 9) || (this.board[endRow+1][startColumn]==-1)){
+                    return true;
+                }
+            }
+        }
+        else {
+            if (Math.abs(endColumn - startColumn)+1 == this.remainingShips[this.remainingShips.length-1]){
+                return true;
+            }
+            if ((startColumn == 0) || (this.board[startRow][startColumn-1]==-1)){
+                if ((endColumn == 9) || (this.board[startRow][endColumn+1]==-1)){
+                    return true;
+                }
+            }
+        }
+        return false
```

Comment:
This is a very complex bit of logic. I think it does have to be this long, given the way it's structured; however, I think you could extract the if statement code (e.g. `(startRow == endRow) && (startColumn == endColumn)` on the top if, and so on) into private methods, if only to give them clear names!